File: kotlin/examples/verify/src/main/kotlin/dev/chatgptoauth/verify/Verify.kt
package dev.chatgptoauth.verify

import dev.chatgptoauth.AuthSession
import dev.chatgptoauth.ResponseContent
import dev.chatgptoauth.ResponseRequest
import dev.chatgptoauth.SubscriptionAI
import dev.chatgptoauth.store.MemoryCredentialStore
import kotlinx.coroutines.runBlocking

/**
 * Signs in and streams one response, to check the port against a live account.
 *
 * ```
 * cd kotlin && ./gradlew :examples:verify:run --console=plain
 * ```
 *
 * The JVM core ships no persistent store — the encrypted one is Android-only — so this signs in
 * every run.
 */
private const val SUBJECT = "example-user"
private const val MODEL = "gpt-5.4-mini"

public fun main(): Unit = runBlocking {
    val auth = AuthSession(store = MemoryCredentialStore())

    val device = auth.startDeviceLogin(SUBJECT)
    println("Open ${device.verificationUrl} and enter code ${device.userCode}\n")
    device.wait()

    val client = SubscriptionAI(auth, SUBJECT)
    val request = ResponseRequest(model = MODEL, input = ResponseContent.Text("Say hello in five words."))
    client.stream(request).collect { event ->
        event.delta?.let { print(it); System.out.flush() }
    }
    println()
}

File: swift/examples/verify/Sources/Verify/Verify.swift
import ChatGPTOAuth
import Foundation

/// Signs in and streams one response, to check the port against a live account.
///
///     cd swift/examples/verify && swift run Verify
///
/// Credentials land in the login Keychain, so a second run skips the sign-in and exercises refresh
/// instead. This is also the only coverage `KeychainCredentialStore` and the device flow have —
/// neither has a unit test.
@main
struct Verify {
    static let subject = "example-user"
    static let model = "gpt-5.4-mini"

    static func main() async throws {
        let store = KeychainCredentialStore(service: "com.openai.chatgpt-oauth.example")
        let auth = AuthSession(store: store)

        if try await auth.status(subject: subject) == nil {
            let device = try await auth.startDeviceLogin(subject: subject)
            print("Open \(device.verificationURL) and enter code \(device.userCode)\n")
            _ = try await device.wait()
        }

        let client = SubscriptionAI(auth: auth, subject: subject)
        let request = ResponseRequest(model: model, input: .text("Say hello in five words."))
        for try await event in client.stream(request) {
            if let delta = event.delta {
                print(delta, terminator: "")
                fflush(stdout)
            }
        }
        print()
    }
}
